### ISSUE TITLE
support `pack` bindings

### ISSUE THREAD

**Issue by taraszka** [dokku/dokku#7119]
### Description of feature

Following https://paketo.io/docs/howto/configuration/#bindings, the current `dokku` implementation lacks `pack` volume bindings.

**Comment by taraszka** [comment]
@josegonzalez, I think volume binding during the build phase may require a completely different approach. What do you think? (This https://dokku.com/docs/advanced-usage/persistent-storage/#build-phase will work partially with `pack`; another thing is that param is required for the `pack` CLI to make it aware.

I am happy to code this feature myself, but I need some guidance, as storage is managed under `storage:<command> <app>,` builder has its own namespace `builder:set`, and the app it owns.

How about `builder-pack:binding <app> <host-dir:container-dir>` in this case?

**Comment by josegonzalez** [comment]
What if we just ignored storage mounts during a pack build _unless_ the container path has the prefix `/platform/bindings/` (or whatever the configured `SERVICE_BINDING_ROOT` env var is), and then pass those along to pack as bindings? That would make it so Dokku doesn't need anything extra and the builder-pack plugin just does the right thing.

**Comment by josegonzalez** [comment]
Yeah the more I think about this, the more I think the existing storage plugin should work fine here. It is possible to set storage specifically for the `build` process and we currently do not filter volumes.

The big issue is that the `pack` binary doesn't support the `-v` shorthand flag (which we use in Dokku, ugh sorry) on the `build` subcommand. This means we'd need to translate them first - probably in the `docker-options#docker-args-build` trigger. Basically split the values and if we see the flag name is `-v`, then change it to `--volume`.

**Comment by taraszka** [comment]
Right, it sounds good to me, too! I'll try to have a look at it during the weekend.

**Comment by josegonzalez** [comment]
Closing as there is a pull request open.

**Comment by turicas** [comment]
Somewhat related: shouldn't it be called `paketo` for the volume permissions? [The current implementation uses packeto](https://github.com/search?q=repo%3Adokku%2Fdokku%20packeto&type=code) (with a "c").

**Comment by josegonzalez** [comment]
Oh yeah probably. Mind making a PR that supports both `paketo` and `packeto`, and maybe does a warning log message if you misspell it?

**Comment by turicas** [comment]
> Oh yeah probably. Mind making a PR that supports both `paketo` and `packeto`, and maybe does a warning log message if you misspell it?

Done in #7587.
